modules/permission-sets/main.tf

```diff
@@ -78,4 +93,18 @@ locals {
   customer_managed_policy_attachments_map = {
     for policy in local.customer_managed_policy_attachments : "${policy.policy_set}.${policy.policy_path}${policy.policy_name}" => policy
   }
+  
+  boundary_policy_map = { for ps in var.permission_sets : ps.name => ps.boundary_policy_attachments if length(ps.boundary_policy_attachments) > 0 }
+  boundary_policy_attachments = flatten([
```

Comment:
Mind including a comment or two explaining your data manipulations here? I think that will be helpful.